src/tools/meson.build

```diff
@@ -0,0 +1,159 @@
+executable('opensc-tool', 'opensc-tool.c', 'util.c',
+    include_directories: core_inc,
+    link_with: [opensc_lib, scconf_lib],
+    install: true,
+)
+
+executable('opensc-explorer', 'opensc-explorer.c', 'util.c',
+    include_directories: core_inc,
+    link_with: [opensc_lib],
+    dependencies: [readline_dep],
+    install: true,
+)
+
+executable('opensc-asn1', 'opensc-asn1.c', 'fread_to_eof.c', 'opensc-asn1-cmdline.c',
+    include_directories: core_inc,
+    link_with: [opensc_lib],
+)
+
+executable('pkcs15-tool', 'pkcs15-tool.c', 'util.c', '../pkcs11/pkcs11-display.c',
+    include_directories: core_inc,
+    link_with: [opensc_lib],
+    dependencies: [openssl_dep],
+    install: true,
+)
+
+executable('pkcs15-crypt', 'pkcs15-crypt.c', 'util.c',
+    include_directories: core_inc,
+    link_with: [opensc_lib],
+    dependencies: [openssl_dep],
+    install: true,
+)
+
+executable('pkcs11-tool', 'pkcs11-tool.c', 'util.c',
+    include_directories: core_inc,
+    link_with: [opensc_lib, pkcs11_lib, pkcs11_mod],
```

Comment:
This prints a deprecation warning because pkcs11_mod is a dlopen() style `shared_module()`.